pkg/bootiso/bootiso.go

```diff
@@ -43,12 +51,56 @@ func ParseConfigFromISO(isoPath string, configType string) ([]boot.OSImage, erro
 	}
 	defer mp.Unmount(0)
 
-	configOpts, err := parseConfigFile(tmp, configType)
+	images, err := parseConfigFile(tmp, configType)
 	if err != nil {
 		return nil, fmt.Errorf("Error parsing config: %v", err)
 	}
 
-	return configOpts, nil
+	return images, nil
+}
+
+// LoadCustomConfigs is an alternative to ParseConfigFromISO that allows us
+// to define the boot parameters ourselves (in a list of Config objects)
+// instead of parsing them from a config file
+func LoadCustomConfigs(isoPath string, configs []Config) ([]boot.OSImage, error) {
+	tmpDir, err := ioutil.TempDir("", "mnt-")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	loopdev, err := loop.New(isoPath, "iso9660", "")
+	if err != nil {
+		return nil, err
+	}
+
+	mp, err := loopdev.Mount(tmpDir, unix.MS_RDONLY|unix.MS_NOATIME)
+	if err != nil {
+		return nil, err
+	}
+	defer mp.Unmount(0)
+
+	var images []boot.OSImage
+	for _, c := range configs {
+		kernelReader := uio.NewLazyFile("file://" + tmpDir + c.KernelPath)
```

Comment:
you should be using path.Join maybe with URL? Not filepath.Join, but path.Join
package main

import "fmt"
import "path"

func main() {
	fmt.Println(path.Join("file://", "x", "y"))

}

but there's a deeper problem here. LazyFiles do not get opened until the first read. 

So the file is not open when you leave this function I believe. 

The mount will unmount the iso.

the open on first read should then fail.

Also, for safety, your deferred functions maybe ought to have some error checking.

Reply:
I was just noticing the problem with the Lazy Files. What would be the better way to open a reader on the kernel and initrd files?